File: servers/javascript/express/src/api/v1/resources/users/router.ts
import Router from 'express-promise-router';
import { createValidator } from '@api/v1/middleware/request-validator';
import { login } from './user.controller';
import { schemas } from './schemas';

const validate = createValidator();

const router = Router();

router.post('/login', validate(schemas.loginSchema), login);

export { router };

File: servers/go/gin/src/api/v1/resources/users/router.go
package users

import (
	"github.com/gin-gonic/gin"
	"github.com/tinadim/polyglot-rest-api/servers/go/gin/src/api/v1/middleware"
	"github.com/tinadim/polyglot-rest-api/servers/go/gin/src/api/v1/resources/users/schemas"
)

func Mount(router gin.RouterGroup) {
	router.POST("/login", middleware.Validate(schemas.GetLoginSchema()), Login)
}
